### ISSUE TITLE
Provide a smarter way to detect & block DNS amplification

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdGuardHome#1665]
First of all, this feature is important only to those who run AGH as a public DNS server.

Currently, we just ship AGH with a relatively small rate limit value by default. This is enough to guarantee that we don't do much harm, but the server stats are distorted anyway.

It'd be better if AGH could automatically block domain names used for amplification.

I suggest using a rather simple algorithm for now:
1. If the client is being constantly rate-limited
2. Most of this client's DNS queries are for the same domain name
3. We should add this domain name to "Disallowed domains" automatically

Users should be able to disable this behavior.

Add a new setting to "DNS settings -> Access settings":

* Name: "Smart DNS amplification detection"
* Description: "Automatically detect and block domain names that are being used for DNS amplification attacks. Please note, that this feature makes sense only if you run AdGuard Home on a public server."

**Comment by shimuldn** [comment]
Waiting for the smarter way.... to protect open dns resolvers.
Thanks

**Comment by ammnt** [comment]
This task also linked with my previous questions: https://github.com/AdguardTeam/AdGuardHome/issues/1660 and https://github.com/AdguardTeam/AdGuardHome/issues/1177

**Comment by Eyeborgs** [comment]
Something similar to what I suggested here #805
But I too like what you've suggested

**Comment by michaelb-ae** [comment]
Actually we are many to use it a "public" because even if we don't spread our url or ip, if we want to use it outside of our home, it become "public" we just hide it's url.
So i hope to see this king of security soon ^^

**Comment by r1bnc** [comment]
Any update?

**Comment by speedster-pta** [comment]
Perhaps adding an option to automatically block any IP that queries a specific domain would help?

**Comment by ameshkov** [comment]
The problem with DNS amplification is that they're smarter now and do not use a single domain name anymore, making the traffic look kinda legitimate.

Your suggestion (as well as mine in the original issue) would work okay against stupid DNS amplification only.

**Comment by Eyeborgs** [comment]
> The problem with DNS amplification is that they're smarter now and do not use a single domain name anymore, making the traffic look kinda legitimate.

Вы не много не правы: я вам для примера приведу один реальный домен который точно используют для для атаки и проверки нашего DNS эта: pizzaseo.com и так же у меня достаточно много и других доменов через которые проверяют работоспособность моего DNS боты.. По сути да не хватает такой функции который предложил [speedster-pta](https://github.com/AdguardTeam/AdGuardHome/issues/1665#issuecomment-1030631824).. Просто я не люблю пользоваться "Неразрешённые домены" потому что я хочу следить за IP ботов и банить их но это все происходит в ручную но хочется специальный раздел в настройках типо "Бан IP по домену" и когда пользователь запросит этот домен его IP автоматически попадет в "Запрещённые клиенты".. Вот какие домены я бы добавил для бана: version.bind, hostname.bind и pizzaseo.com и другие.. 

Простите что я на Русском но через переводчик возможно могут быть искажения..

**Comment by speedster-pta** [comment]
> The problem with DNS amplification is that they're smarter now and do not use a single domain name anymore, making the traffic look kinda legitimate.
> 
> Your suggestion (as well as mine in the original issue) would work okay against stupid DNS amplification only.

Fair enough, but at least it's a start and should be relatively easy to implement.

**Comment by xlionjuan** [comment]
Any updates? Recently I got massive queries that querying `fusu.cc` and `cloudflare.com`'s TXT records, so I manually added those to my blocklist, but it still needs human to take actions.

```
||cloudflare.com^$dnstype=TXT
||fusu.cc^$dnstype=TXT
```

**Comment by ltwc114** [comment]
> Any updates? Recently I got massive queries that querying 和 's TXT records, so I manually added those to my blocklist, but it still needs human to take actions.`fusu.cc``cloudflare.com`
> 
> ```
> ||cloudflare.com^$dnstype=TXT
> ||fusu.cc^$dnstype=TXT
> ```

You can add domain names that need to be blocked in Settings - DNS Settings - Access Settings - Forbidden Domain Names.

**Comment by xlionjuan** [comment]
> > Any updates? Recently I got massive queries that querying 和 's TXT records, so I manually added those to my blocklist, but it still needs human to take actions.`` fusu.cc``cloudflare.com ``
> > ```
> > ||cloudflare.com^$dnstype=TXT
> > ||fusu.cc^$dnstype=TXT
> > ```
> 
> You can add domain names that need to be blocked in Settings DNS Settings Access Settings Forbidden Domain Names

This is not automatic.
